web/src/screens/private/Schedules/AddSchedule.js

```diff
@@ -0,0 +1,433 @@
+import { FormControl, FormControlLabel } from 'material-ui/Form';
+import { InputLabel } from 'material-ui/Input';
+import { MenuItem } from 'material-ui/Menu';
+import { CircularProgress } from 'material-ui/Progress';
+import Select from 'material-ui/Select';
+import Snackbar from 'material-ui/Snackbar';
+import Stepper, { Step, StepLabel, StepContent } from 'material-ui/Stepper';
+import Switch from 'material-ui/Switch';
+import TextField from 'material-ui/TextField';
+import Typography from 'material-ui/Typography';
+
+import CREATE_SCHEDULE_MUTATION from '../../../graphql/create-schedule.mutation';
+import CURRENT_USER_QUERY from '../../../graphql/current-user.query';
+
+import styles from './AddSchedule.styles';
+
+function getSteps() {
+  return ['Select group', 'Specify details', 'Capabilities', 'Enter dates'];
+}
+
+function getStepContent(step) {
+  switch (step) {
+    case 0:
+      return 'Who will the request target?';
+    case 1:
+      return 'Enter a meaningful title and details for the request. Priority does nothing yet.';
+    case 2:
+      return 'What capabilities are required for the request?';
+    case 3:
+      return 'Dates can be an ongoing request with no end date or within a specified date range. e.g. a request for a weekend weather event.';
+    default:
+      return 'Unknown step';
+  }
+}
+
+class AddSchedule extends React.Component {
+  state = {
+    open: false,
+    message: '',
+    name: '',
+    groupId: '',
+    details: '',
+    priority: 2,
+    capability: '',
+    startTime: moment()
+      .add(1, 'day')
+      .startOf('day')
+      .format('YYYY-MM-DD[T]HH:mm'),
+    endTime: moment()
+      .add(3, 'day')
+      .startOf('day')
+      .format('YYYY-MM-DD[T]HH:mm'),
+    activeStep: 0,
+    useDates: false,
+  };
+
+  onGroupChange = (e) => {
+    const groupId = e.target.value;
+    this.setState(() => ({ groupId }));
+  };
+
+  onPriorityChange = (e) => {
+    const priority = e.target.value;
+    this.setState(() => ({ priority }));
+  };
+
+  onCapabilityChange = (e) => {
+    const capability = e.target.value;
+    this.setState(() => ({ capability }));
+  };
+
+  onNameChange = (e) => {
+    const name = e.target.value;
+    this.setState(() => ({ name }));
+  };
+
+  onDetailsChange = (e) => {
+    const details = e.target.value;
+    this.setState(() => ({ details }));
+  };
+
+  onStartTimeChange = (e) => {
+    const startTime = e.target.value;
+    this.setState(() => ({ startTime }));
+  };
+
+  onEndTimeChange = (e) => {
+    const endTime = e.target.value;
+    this.setState(() => ({ endTime }));
+  };
+
+  onUseDatesChange = () => {
+    const useDates = !this.state.useDates;
+    this.setState(() => ({ useDates }));
+  };
+
+  handleNext = () => {
+    const steps = getSteps();
+    if (this.state.activeStep === steps.length - 1) {
+      const { name, details, useDates, startTime, endTime, groupId } = this.state;
+      let startTimeUnix = 0;
```

Comment:
i'd really prefer these were constants somewhere even if it's not the same file as the client/server.

Reply:
I agree - I'll fix up